R/run_calculate_loanbook_coverage.R

```diff
@@ -40,12 +44,20 @@ run_calculate_loanbook_coverage <- function(config) {
   matched_prioritized <- readr::read_csv(
     file = file.path(dir_matched, list_matched_prioritized),
     col_types = col_types_matched_prioritized,
-    col_select = dplyr::all_of(col_select_matched_prioritized)
+    col_select = dplyr::all_of(c(by_group, col_select_matched_prioritized))
   )
 
+  # add helper column to facilitate calculation of meta results----
+  # TODO: decide if this should be removed from outputs
+  if (is.null(by_group)) {
+    by_group <- "meta"
+    matched_prioritized <- matched_prioritized %>%
+      dplyr::mutate(meta = "meta")
+  }
+
   matched_companies <- matched_prioritized %>%
     dplyr::distinct(
-      .data$group_id,
+      !!rlang::sym(by_group),
```

Comment:
nit: is this less complicated?
```suggestion
      .data[[by_group]],
```

Reply:
not 100% sure my suggestion works

Reply:
it does, I thought I had tried this, but I must have been doing something else